src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -81,4 +114,192 @@ private bool StripAssembly(ITaskItem assemblyItem)
         return true;
     }
 
+    private bool TrimMethods(ITaskItem assemblyItem)
+    {
+        string methodTokenFile = assemblyItem.GetMetadata("MethodTokenFile");
+        if (string.IsNullOrEmpty(methodTokenFile))
+        {
+            Log.LogError($"Metadata MethodTokenFile of {assemblyItem.ItemSpec} is empty");
+        }
+        if (!File.Exists(methodTokenFile))
+        {
+            Log.LogMessage(MessageImportance.Low, $"{methodTokenFile} doesn't exist.");
+            return true;
+        }
+
+        using StreamReader sr = new(methodTokenFile);
+        {
+            string? assemblyFilePath = sr.ReadLine();
+            if (string.IsNullOrEmpty(assemblyFilePath))
+            {
+                return true;
+            }
+
+            if (!File.Exists(assemblyFilePath))
+            {
+                Log.LogError($"{assemblyFilePath} read from {methodTokenFile} doesn't exist.");
+                return true;
+            }
+
+            string trimmedAssemblyFilePath = ComputeTrimmedAssemblyPath(assemblyFilePath);
+            bool isTrimmed = false;
+
+            using FileStream fs = File.Open(assemblyFilePath, FileMode.Open);
+            {
+                using PEReader peReader = new(fs, PEStreamOptions.LeaveOpen);
+                MetadataReader mr = peReader.GetMetadataReader();
+
+                string guidValue = ComputeGuid(mr);
+                string? expectedGuidValue = sr.ReadLine();
+                if (!string.Equals(guidValue, expectedGuidValue, StringComparison.OrdinalIgnoreCase))
+                {
+                    Log.LogError($"[ILStrip] GUID value of {assemblyFilePath} doesn't match the value listed in {methodTokenFile}.");
+                    return true;
```

Comment:
Shouldn't this be stopping the task? Use `throw new LogAsErrorException`, so it's clearer what this `return true` will really do.

Reply:
This is a nice-to-have feature. If GUID doesn't match, it won't be trimmed. It shouldn't stop other assemblies with matching GUID to be trimmed.